network/no_ingress_conn_alert.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package network
+
+import (
+	"errors"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	"github.com/ava-labs/avalanchego/utils/constants"
+)
+
+// ErrNoIngressConnections denotes that no node is connected to this validator.
+var ErrNoIngressConnections = errors.New("no ingress connections")
+
+type ingressConnectionCounter interface {
+	IngressConnCount() int
+}
+
+type validatorRetriever interface {
+	GetValidator(subnetID ids.ID, nodeID ids.NodeID) (*validators.Validator, bool)
+}
+
+type noIngressConnAlert struct {
+	selfID             ids.NodeID
+	ingressConnections ingressConnectionCounter
+	validators         validatorRetriever
+}
+
+func ingressConnResult(n int, areWeValidator bool) map[string]interface{} {
+	return map[string]interface{}{"ingressConnectionCount": n, "primary network validator": areWeValidator}
+}
+
+func (nica *noIngressConnAlert) checkHealth() (interface{}, error) {
```

Comment:
Would it make more sense for this to just be a function that takes in:
```
	selfID             ids.NodeID
	ingressConnections ingressConnectionCounter
	validators         validatorRetriever
```
rather than defining the whole `noIngressConnAlert` struct?

Reply:
yes, in the previous incarnations this struct had state, so it had to be one. Now it can be a function as you pointed out. Fixed [here](https://github.com/ava-labs/avalanchego/pull/3719/commits/9c9ca56e1ad115a5e5ba025a4df288e978e4ec12#diff-a717b2d70449f71cde25c97e15008b7a6f8b21c9be34bb89c16d77a871546b20R25).